### PR TITLE
chore: bump OT core to v0.48.0

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: otelcolbuilder/.otelcol-builder.yaml [hunk 1]
```diff
@@ -197,23 +197,23 @@ replaces:
   # github.com/open-telemetry/opentelemetry-log-collection/pipeline.Config.Build({{0x9aeb030, 0xc00074a090}, {0xc00071c510, 0x3, 0x3}}, 0x2?)
   #         github.com/open-telemetry/opentelemetry-log-collection@v0.26.0/pipeline/config.go:83 +0x4c5 fp=0xc0005c4c98 sp=0xc0005c4b28 pc=0x7212ee5
   # github.com/open-telemetry/opentelemetry-collector-contrib/internal/stanza.createLogsReceiver.func1({0x0?, 0xffffffffff?}, {{0xc0000f5d40, {0x9a92fc0, 0xc000b250a0}, {0x9a9e020, 0xca100b8}, 0x0}, {{0x8fe90d1, 0xc}, ...}}, ...)
-  #         github.com/open-telemetry/opentelemetry-collector-contrib/internal/stanza@v0.47.0/factory.go:81 +0x57d fp=0xc0005c4fb8 sp=0xc0005c4c98 pc=0x724a89d
+  #         github.com/open-telemetry/opentelemetry-collector-contrib/internal/stanza@v0.48.0/factory.go:81 +0x57d fp=0xc0005c4fb8 sp=0xc0005c4c98 pc=0x724a89d
```
Comment (pmalek-sumo) [inline otelcolbuilder/.otelcol-builder.yaml:200]: I'd leave those comments as-is

Comment (astencel-sumo) [reply otelcolbuilder/.otelcol-builder.yaml]: Reverted to the original version of the comments, with `0.46.0`.

**Review by pmalek-sumo — APPROVED** [review]

File: otelcolbuilder/.otelcol-builder.yaml [hunk 2]
```diff
@@ -197,23 +197,23 @@ replaces:
...
   # go.opentelemetry.io/collector/service.newService(0xc000fe3980)
-  #         go.opentelemetry.io/collector@v0.47.0/service/service.go:74 +0x5ce fp=0xc0005c58a0 sp=0xc0005c5680 pc=0x7be8dae
+  #         go.opentelemetry.io/collector@v0.46.0/service/service.go:74 +0x5ce fp=0xc0005c58a0 sp=0xc0005c5680 pc=0x7be8dae
   # go.opentelemetry.io/collector/service.(*Collector).setupConfigurationComponents(0xc00016e460, {0x9acc778, 0xc00012a000})
-  #         go.opentelemetry.io/collector@v0.47.0/service/collector.go:200 +0x31b fp=0xc0005c5a28 sp=0xc0005c58a0 pc=0x7be679b
+  #         go.opentelemetry.io/collector@v0.46.0/service/collector.go:200 +0x31b fp=0xc0005c5a28 sp=0xc0005c58a0 pc=0x7be679b
   # go.opentelemetry.io/collector/service.(*Collector).Run(0xc00016e460, {0x9acc778, 0xc00012a000})
-  #         go.opentelemetry.io/collector@v0.47.0/service/collector.go:240 +0x24e fp=0xc0005c5b48 sp=0xc0005c5a28 pc=0x7be6aae
+  #         go.opentelemetry.io/collector@v0.46.0/service/collector.go:240 +0x24e fp=0xc0005c5b48 sp=0xc0005c5a28 pc=0x7be6aae
   # go.opentelemetry.io/collector/service.NewCommand.func1(0xc000a52280, {0x8fd4b1d?, 0x2?, 0x2?})
```
Comment (sumo-drosiek) [inline otelcolbuilder/.otelcol-builder.yaml:217]: We shouldn't change that

**Review by sumo-drosiek — APPROVED** [review]

File: otelcolbuilder/.otelcol-builder.yaml (hunk 2 above)
Comment (pmalek-sumo) [reply otelcolbuilder/.otelcol-builder.yaml:217]: This was the original version I believe. In any case: I've tested this and we don't need this anymore after bump to 0.48.0.

Comment (astencel-sumo) [reply otelcolbuilder/.otelcol-builder.yaml:217]: Exactly, that's why I changed it to the original. I assume this shouldn't have been from 0.46.0 to 0.47.0 in the first place.

**Pull request merged**